torch/distributed/_cuda_p2p/__init__.py

```diff
@@ -404,6 +409,14 @@ def shard_consumer(shard: torch.Tensor, rank: int) -> None:
     return unflatten(ag_out), [unflatten(output) for output in outputs]
 
 
+def get_optimal_A_shard_stride_order_for_fused_all_gather_matmul(
```

Comment:
Can we refactor this to an utility that given a contiguous tensor, converts it to a tensor with a given permuted layout?

I think something along the lines of
```
def restride_tensor(x, perm):
    return x.permute(perm).contiguous().permute(inv_perm)
```
would work.

Reply:
Do I need the `needs_fixed_stride_order` tag on the operator for this approach to work?

Reply:
You can still do it within Inductor, just a suggestion on how to organize the code a bit nicer.